tests/unit/helpers/test_api_client.py

```diff
@@ -2,57 +2,386 @@
+        )
+        api_client.request_server = pretend.call_recorder(
+            lambda *a, **kw: pretend.stub(status_code=200)
+        )
+
+        result = api_client.get_headers(
+            {"SERVER": "http://server", "TOKEN": "fake_token"}
+        )
+
+        assert result == {"Authorization": "Bearer fake_token"}
+        assert api_client.is_logged.calls == [
+            pretend.call("http://server", "fake_token")
+        ]
+        assert api_client.request_server.calls == [
+            pretend.call(
+                "http://server",
+                api_client.URL.bootstrap.value,
+                api_client.Methods.get,
+                headers={"Authorization": "Bearer fake_token"},
+            )
+        ]
+
+    def test_get_headers_never_logged(self):
+        with pytest.raises(api_client.click.ClickException) as err:
+            api_client.get_headers({})
+
+        assert "Login first. Run 'rstuf admin login'" in str(err)
+
+    def test_get_headers_is_logged_state_false(self):
+        api_client.is_logged = pretend.call_recorder(
+            lambda *a, **kw: api_client.Login(
+                state=False, data={"expired": False}
+            )
+        )
+
+        with pytest.raises(api_client.click.ClickException) as err:
+            api_client.get_headers(
+                {"SERVER": "http://server", "TOKEN": "fake_token"}
+            )
+
+        assert "re-login" in str(err)
+        assert api_client.is_logged.calls == [
+            pretend.call("http://server", "fake_token")
+        ]
+
+    def test_get_headers_is_logged_state_true_expired_token(self):
+        api_client.is_logged = pretend.call_recorder(
+            lambda *a, **kw: api_client.Login(
+                state=True, data={"expired": True}
+            )
+        )
+
+        with pytest.raises(api_client.click.ClickException) as err:
+            api_client.get_headers(
+                {"SERVER": "http://server", "TOKEN": "fake_token"}
+            )
+
+        assert "The token has expired" in str(err)
+        assert api_client.is_logged.calls == [
+            pretend.call("http://server", "fake_token")
+        ]
+
+    def test_get_headers_unexpected_error(self):
+        api_client.is_logged = pretend.call_recorder(
+            lambda *a, **kw: api_client.Login(
+                state=True, data={"expired": False}
+            )
+        )
+        api_client.request_server = pretend.call_recorder(
+            lambda *a, **kw: pretend.stub(status_code=500, text="error body")
+        )
+        with pytest.raises(api_client.click.ClickException) as err:
+            api_client.get_headers(
+                {"SERVER": "http://server", "TOKEN": "fake_token"}
+            )
+
+        assert "Unexpected error" in str(err)
+        assert api_client.is_logged.calls == [
+            pretend.call("http://server", "fake_token")
+        ]
+
+    def test_task_status(self):
+        fake_json = Mock()
+        fake_json.side_effect = [
```

Comment:
I am unfamiliar with `Mock()` objects.
What does `side_effect` do and why you decided to use a `Mock` object?

Reply:
Unfortunately, pretend can't make multiple different mocked retuvariousmultiple calls. I'm even thinking of submitting a PR to implement it there.

`Mock()` creates a mocked object and the `side_effect` gives a sequence of mocked returns (from a List) or even an `Exception`. [See the documentation.](https://docs.python.org/3/library/unittest.mock.html#quick-guide)
As the implementation does, `def task_status` calls multiple times to the REST API tasks. 

https://github.com/vmware/repository-service-tuf-cli/blob/353f44b79060546a4a976080ce940e11af5ba92a/repository_service_tuf/helpers/api_client.py#L117-L120

I simulate multiple returns with different responses, as it is in the actual call with different statuses. I test the logic unit from the implementation.

Reply:
That's an interesting use case. Thanks for sharing that info, it will be useful.